gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/cockpit/command/handler/DeployModelCommandHandler.java

```diff
@@ -90,7 +90,7 @@ public DeployModelCommandHandler(
 
             ApiEntityResult result;
 
-            if (apiService.exists(apiId)) {
+            if (apiService.findByEnvironmentIdAndCrossId(environmentId, apiId).isPresent()) {
```

Comment:
Does apiId refers to a crossId here ? If that the case I think we need to map the result to the actual apiId as subsequent method calls expect this ID and not the cross ID

Reply:
Done in the fixup commit